src/ReinforcementLearningCore/src/policies/agent/agent_srt_cache.jl

```diff
@@ -1,65 +1,49 @@
 import Base.push!
 
-mutable struct SRT{S,R,T}
+mutable struct SA{S,A}
```

Comment:
I do not get why we want to refactor the agent_cache here? The PR is about when to optimise. In my opinion, this issue (if it is actually one) has to be addressed in another PR.